tests/annotations/tests.py

```diff
@@ -148,6 +148,19 @@ def test_mixed_type_annotation_numbers(self):
         combined = int(test.pages + test.rating)
         self.assertEqual(b.combined, combined)
 
+    def test_empty_expression_annotation(self):
+        books = Book.objects.annotate(
+            selected=ExpressionWrapper(Q(pk__in=[]), output_field=BooleanField())
```

Comment:
We'll revert the changes in `django/db/models/sql/compiler.py` as well then? This is acting as a regression test for those changes.